akka-http-core/src/main/scala/akka/http/impl/settings/HostConnectionPoolSetup.scala

```diff
@@ -4,6 +4,9 @@
 
 package akka.http.impl.settings
 
+import akka.annotation.InternalApi
+
 /** INTERNAL API */
+@InternalApi
 final case class HostConnectionPoolSetup(host: String, port: Int, setup: ConnectionPoolSetup)
```

Comment:
double check this one, I think we may actually surface it in some places in user api?